openrr-teleop/src/control_nodes_config.rs

```diff
@@ -115,6 +154,230 @@ impl ControlNodesConfig {
             }
         }
 
-        nodes
+        Ok(nodes)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use arci::{
+        DummyJointTrajectoryClient, DummyLocalization, DummyMoveBase, DummyNavigation, DummySpeaker,
+    };
+    use openrr_client::{
+        create_random_jacobian_ik_solver, IkSolverParameters, IkSolverWithChain,
+        OpenrrClientsConfig,
+    };
+
+    use super::*;
+    use crate::{default_move_step_angular, default_move_step_linear, default_step_duration_secs};
+
+    #[test]
+    fn test_gen() {
+        // JoyJointTeleopConfig
+        let config = JoyJointTeleopNodeConfig {
+            mode: String::from("a"),
+            joint_step: 1.0_f64,
+            step_duration_secs: default_step_duration_secs(),
+        };
```

Comment:
I  agree but why I  can use `toml style`？

These default function have  what purpose？I think using public(`pub(crate)` is wrong, I  want to  write `pub`. Sorry.).

Reply:
You can use toml style because `IkNodeConfig` struct implements `Deserialize` trait.
https://github.com/openrr/openrr/blob/33a1ba24b925b5a79f382c011bd236c169e8b1fe/openrr-teleop/src/ik.rs#L233
(I have shown it as an example, but do not recommend it.
You can use any values for the tests.)

Those default functions are only for providing default values on deserializing.
(cf. https://serde.rs/attr-default.html)
And they are not intended to be used publicly.

Reply:
Thank you for explanation.
This problem solve by to change to using specific value and returning private function.